src/include/planner/operator/sip/side_way_info_passing.h

```diff
@@ -5,13 +5,58 @@
 namespace kuzu {
 namespace planner {
 
-enum class SidewaysInfoPassing : uint8_t {
+enum class SemiMaskPosition : uint8_t {
     NONE = 0,
-    PROHIBIT = 1,
-    PROBE_TO_BUILD = 2,
-    PROHIBIT_PROBE_TO_BUILD = 3,
-    BUILD_TO_PROBE = 4,
-    PROHIBIT_BUILD_TO_PROBE = 5,
+    ON_BUILD = 1,
+    ON_PROBE = 2,
+    PROHIBIT = 3,
+};
+
+/*
+ * Side way information passing create dependency between pipelines.
+ * */
+enum class SIPDependency : uint8_t {
+    NONE = 0,
+    PROBE_DEPENDS_ON_BUILD = 1,
+    BUILD_DEPENDS_ON_PROBE = 2,
+};
+
+/*
+ * Direction of side way information passing. If direction is probe to build, then probe side
+ * must have been materialized and we need to construct a new pipeline to scan materialized result.
+ * */
+enum class SIPDirection {
+    NONE = 0,
+    PROBE_TO_BUILD = 1,
+    BUILD_TO_PROBE = 2,
+};
+
+/*
+ * We perform side way information passing in different scenarios
+ * 1. Semi mask
+ *
+ * During join order enumeration, we might disable semi mask if probe cardinality is large to avoid
+ * large materialization.
+ *
+ * During filter push down, we disable semi mask if join condition is not id-based
```

Comment:
I'm a bit lost with the comment. Are you enumerating cases when we do not use semi-mask? If so, have a sentence that says something like: "We do not use semi-mask-based sip for every join we make. Below are several cases when semi-mask-based sip is disabled: - .... - ..."